tests/unittests/test_aggregate_counts.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+import pandas
+
+from src.processing import aggregate_counts as aggregate_counts
+from src.utils import spark as spark_utils
+from pyspark.sql import functions as F
+from pyspark.sql import SparkSession
+
+def test_get_aggregate_counts():
+    """
+    Tests get_aggregate_counts
+    """
+    spark = spark_utils.create_spark_session('tests')
+
+    expected_data = [
+        (3,),
+    ]
+    expected_cols = ['count']
+    df_expected = spark.createDataFrame(expected_data, expected_cols)
+
+    unaggregated_data = [
+        ('group_1',),
+        ('group_2',),
+        ('group_2',),
+        ('group_3',),
+        ('group_3',),
+        ('group_3',),
+    ]
+    unaggregated_cols = ['group_name']
+    df_unaggregated = spark.createDataFrame(unaggregated_data, unaggregated_cols)
+
+    df_actual = aggregate_counts.get_aggregate_counts(df_unaggregated, 'group_name', 'count')
+
+    assert df_actual.toPandas().equals(df_expected.toPandas())
+    
+
+def test_get_grouped_aggregate_counts():
+    """
+    Tests get_aggregate_counts
+    """
+    spark = spark_utils.create_spark_session('tests')
+
+    expected_data = [
+        ('group_1', 1),
+        ('group_2', 1),
+        ('group_3', 2),
```

Comment:
I found this a bit unintuitive at first - as I expect it to "sum" the counts together! when instead the function is just doing a  count distinct! not sure if anything needs changing - just thought I'd feed that back!